core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/StoreCallbacksImpl.scala

```diff
@@ -320,6 +320,9 @@ object StoreCallbacksImpl extends StoreCallbacks with Logging with Serializable
 
   override def shouldStopRecovery(): Boolean =
     MemoryManagerCallback.memoryManager.shouldStopRecovery()
+
+  override def isEnterpriseEdition: Boolean =
+    SnappySession.isEnterpriseEdition
 }
```

Comment:
This was added to enable GemFireCacheImpl check for enterprise edition. It cannot access SnappyDataVersion.isEnterpriseEdition as it is in snappy-cluster and the ClusterCallbacks is available in gemfirexd module which gemfire module cannot access.

So added the check in SnappySession using reflection which this storecallback (it should really be renamed to CoreCallbacks similar to ClusterCallbacks) uses.